test/test_linalg.py

```diff
@@ -1037,6 +1037,96 @@ def test_tensorsolve_errors_and_warnings(self, device, dtype):
         with self.assertRaisesRegex(RuntimeError, "result dtype Int does not match self dtype"):
             torch.linalg.tensorsolve(a, b, out=out)
 
+    def _test_dot_vdot_vs_numpy(self, device, dtype, torch_fn, np_fn):
+        def compare_with_numpy_bin_op(torch_fn, np_fn, x, y):
+            y_np = y.cpu().numpy()
+
+            # `compare_with_numpy` takes care of moving `x` to correct device for calling np_fn.
+            self.compare_with_numpy(lambda inp: torch_fn(inp, y), lambda inp: np_fn(inp, y_np), x)
+
+            # Test out variant
+            ref = torch_fn(x, y)
+            out = torch.zeros_like(ref)
+            torch_fn(x, y, out=out)
+            self.assertEqual(ref, out)
+
+        # Use this tensor for out variant tests.
+        out = torch.randn((), dtype=dtype, device=device)
+
+        def compare_out_variant(torch_fn, x, y):
```

Comment:
Is this redundant with lines 1050-1051 above?

Reply:
It is, I'm keeping only this one.